### PR TITLE
MGMT-7742: Fixing host getting stuck in "binding" state

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Once Bind command isssued, the host moves to Binding state and its ClusterID is set
to the target cluster. So, from instruction manager POV this host is now bound host and
not unbound host. The fix is to move binding instuction from unboind host instruction manager
commands set, to the bound host instruction manager command set.
In addition fixing the security definitions for Bind (adn unbind) command. It should be user
auth, and not agent auth

# Assisted Pull Request

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/host/hostcommands/instruction_manager_test.go [hunk 1]
```diff
@@ -251,10 +255,11 @@ var _ = Describe("instruction_manager", func() {
 				models.StepTypeInventory,
 			})
 		})
-
-		It("binding", func() {
-			checkStep(models.HostStatusBinding, nil)
-		})
+		/*
```
Comment (rollandf) [inline internal/host/hostcommands/instruction_manager_test.go:258]: Please delete commented code

Comment (yevgeny-shnaidman) [reply internal/host/hostcommands/instruction_manager_test.go]: fixed

File: internal/host/hostcommands/instruction_manager.go [hunk 2]
```diff
@@ -117,7 +118,6 @@ func NewInstructionManager(log logrus.FieldLogger, db *gorm.DB, hwValidator hard
 			models.HostStatusDisabledUnbound:     {[]CommandGetter{}, defaultBackedOffInstructionInSec, models.StepsPostStepActionContinue},
 			models.HostStatusInsufficientUnbound: {[]CommandGetter{inventoryCmd}, defaultNextInstructionInSec, models.StepsPostStepActionContinue},
 			models.HostStatusKnownUnbound:        {[]CommandGetter{inventoryCmd}, defaultNextInstructionInSec, models.StepsPostStepActionContinue},
-			models.HostStatusBinding:             {[]CommandGetter{noopCmd}, 0, models.StepsPostStepActionExit},
```
Comment (filanov) [inline internal/host/hostcommands/instruction_manager.go:120]: Why did you remove this part? 
How will day2 works with late binding? 
We have a host registered to InfraEnv now we bind it to a day2 cluster what will be the flow?

Comment (yevgeny-shnaidman) [reply internal/host/hostcommands/instruction_manager.go:120]: it is the same flow as day1. We need to add a Binding state to addHostsClusterToSteps instruction manager set (MGMT-7757)

Comment (yevgeny-shnaidman) [reply internal/host/hostcommands/instruction_manager.go:120]: in any case, there can not be state Binding for unbound host, since once host is moved to stateBinding, it becomes Bound host (no matter if day1 or day2
